pkg/reconciler/revision/revision.go

```diff
@@ -93,16 +98,22 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) (boo
 	}
 
 	logger := logging.FromContext(ctx)
-	statuses, err := c.resolver.Resolve(logger, rev, opt, cfgs.Deployment.RegistriesSkippingTagResolving, cfgs.Deployment.DigestResolutionTimeout)
+	initContainerStatuses, statuses, err := c.resolver.Resolve(logger, rev, opt, cfgs.Deployment.RegistriesSkippingTagResolving, cfgs.Deployment.DigestResolutionTimeout)
 	if err != nil {
 		// Clear the resolver so we can retry the digest resolution rather than
 		// being stuck with this error.
 		c.resolver.Clear(types.NamespacedName{Namespace: rev.Namespace, Name: rev.Name})
 		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing, err.Error())
 		return true, err
 	}
-	if len(statuses) > 0 {
-		rev.Status.ContainerStatuses = statuses
+
+	if len(statuses) > 0 || len(initContainerStatuses) > 0 {
+		if len(statuses) > 0 {
+			rev.Status.ContainerStatuses = statuses
+		}
+		if len(initContainerStatuses) > 0 {
+			rev.Status.InitContainerStatuses = initContainerStatuses
+		}
```

Comment:
any reason we can't assign in either case? (given nil array and empty array should be the same when serialised I think, and we already checked that one of status/initStatus is non-nil so the resolve happened); i.e:

```suggestion
		rev.Status.ContainerStatuses = statuses
		rev.Status.InitContainerStatuses = initContainerStatuses
```

Reply:
In the past we had: 

```
	if len(statuses) > 0 {
		rev.Status.ContainerStatuses = statuses
		return true, nil
	}
```
So I wanted to keep the same semantics.